networkd/network_server.ml

```diff
@@ -1012,13 +1013,31 @@ module PVS_proxy = struct
     open S.PVS_proxy
 
     let path = ref "/opt/citrix/pvsproxy/socket/pvsproxy"
-
-	let do_call call =
-		try
-			Jsonrpc_client.with_rpc ~path:!path ~call ()
-		with e ->
-			error "Error when calling PVS proxy: %s" (Printexc.to_string e);
-			raise PVS_proxy_connection_error
+    
+    let do_call call =
+            let timebox ~timeout f =
+                let fd_in, fd_out = Unix.pipe () in
+                let _ =  Thread.create
+                    (fun () ->
+                    f;
+                    debug "Complete execution in thread.";
+                    Unix.close fd_out) () in
+                if Thread.wait_timed_read fd_in timeout then
+                    debug "Timed execution completion."
+                else
+                    debug "Timed execution timed out.";
+                Unix.close fd_in
+            in
+            
+            let rpc_call =
+                try
+                    Jsonrpc_client.with_rpc ~path:!path ~call ()
```

Comment:
Removing the `()` at the end would be a good start. Otherwise (as @mseri said), this `rpc_call` won't be a function, the this would block here immediately before even creating any threads.